### ISSUE TITLE
Conflict with Local to Global Option Set Converter plugin

### ISSUE THREAD

**Issue by rajyraman** [mariusso/XrmToolboxPlugins#1]
I have dll for Local to Global Option Set Converter in the Plugin folder along with Plugin Retriever. When this is the case, an error is displayed when trying to connect to a CRM org. There seems to be some sort of conflict between these two plugins. Is there anyway to workaround this issue?

![errorpluginretriever](https://cloud.githubusercontent.com/assets/5035266/7505325/b97581e4-f494-11e4-945c-854f28848562.png)

**Comment by MscrmTools** [comment]
It seems the problem occurs when two plugins embed Early Bound entities

**Comment by mariusso** [comment]
Yes, Nice catch. It is indeed a problem related to early bound. I haven't figured out what to do yet. As you can see, the type names are different. I have also tried renaming the classes themselves. But the problem is not with the partial classes. Each early bound generated class gets an attribute called EntityLogicalNameAttribute. This is the problem:

![image](https://cloud.githubusercontent.com/assets/7030002/7510629/9bcba6fa-f49e-11e4-993a-804cad70c1a6.png)

I don't know what is does, and I haven't tried editing it. Each class already have public const string with the entity logical name.

The temporary solution is to remove the other dll's temporarily. 

For a more permanent solution, I have to figure out this problem or just use late bound.

**Comment by mariusso** [comment]
Any suggestions? @MscrmTools

**Comment by MscrmTools** [comment]
Could you try to reference the use of PluginAssembly class as YourNamespace.PluginAssembly in your code. If you keep the full name of the class, maybe the conflict won't occur because .NEt will know exactly what class to use

**Comment by mariusso** [comment]
@MscrmTools I found [this.](http://www.infinite-x.net/2013/07/17/error-proxy-type-with-the-name-account-has-been-defined-by-another-assembly/)

Can I modify the CrmConnection in my plugin?

**Comment by MscrmTools** [comment]
With some changes, yes. 
When the plugin is loaded, some parameters are passed to method UpdateConnection. Two of them are the IOrganizationService and the ConnectionDetail (a class from XrmToolBox). This ConnectionDetail object exposes a method GetCrmConnectionString that allows you to recreate a new CrmConnection. From this CrmConnection object, you can override the IOrganizationService that was passed from XrmToolBox.

Clear enough?

**Comment by mariusso** [comment]
@MscrmTools I'll give it at shot. Thanks!

**Comment by MscrmTools** [comment]
I could also update XrmToolBox to include this process in PluginBase class but that would mean to wait for next release for a fix
So, if it works for you, I will include it in a futur release
